internal/pkg/otel/config/config.go

```diff
@@ -0,0 +1,48 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package config
+
+import (
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent/internal/pkg/config"
+	"github.com/elastic/elastic-agent/internal/pkg/otel/manager"
+)
+
+var executionMode = manager.EmbeddedExecutionMode
```

Comment:
Does this really need to be a global? Could we find a way of not making this a global and instead pass this information around.

Reply:
ok valid point and I am not in favour of globals, so lemme check if this can be altered (I think it does)

Reply:
Thanks for changing it.